lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,355 @@
+        Get request for querying a metadata session of a user given a metadata id.
+        This requires a Pbench auth token in the header field
+
+        The url requires a metadata session id such as /user/metadata/<int:id>
+
+        Required headers include
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success",
+                    "data": {
+                        "id": "metadata_id",
+                        "value": "client text blob"
+                        "created": "Datetime string"
+                        "updated": "Datetime String"
+                        "key": "key string"
+                        }
+                }
+        """
+        if not id:
+            self.logger.warning("Metadata id not provided during metadata query")
+            abort(400, message="Please provide a metadata id to query")
+
+        try:
+            # Fetch the metadata session
+            metadata_session = Metadata.query(id=id)[0]
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in the GET request while querying the Metadata model, id: {}",
+                id,
+            )
+            abort(500, message="INTERNAL ERROR")
+
+        # Verify if the metadata session id in the url belongs to the logged in user
+        self.verify_metadata(metadata_session, "get")
+
+        response_object = {
+            "message": "success",
+            "data": metadata_session.get_json(),
+        }
+        return make_response(jsonify(response_object), 200)
+
+    @Auth.token_auth.login_required()
+    def put(self, id=None):
+        """
+        Put request for updating a metadata session of a user given a metadata id.
+        This requires a Pbench auth token in the header field
+
+        The url requires a metadata session id such as /user/metadata/<int:id>
+
+        This requires a JSON data with required user metadata fields to update
+        {
+            "value": "new text value",
+            ...
+        }
+        To update the value field, it is required to send the complete text blob again
+
+        Required headers include
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success",
+                    "data": {
+                        "id": "metadata_id",
+                        "value": "client text blob"
+                        "created": "Datetime string"
+                        "updated": "Datetime String"
+                        "key": "key string"
+                        }
+                }
+        """
+        if not id:
+            self.logger.warning("Metadata id not provided during metadata query")
+            abort(400, message="Please provide a metadata id to query")
```

Comment:
In this case, your value/row id is "to change/update". Saying "query" sounds potentially confusing.